cmd/tailscaled/tailscaled.go

```diff
@@ -153,6 +154,14 @@ var subCommands = map[string]*func([]string) error{
 
 var beCLI func() // non-nil if CLI is linked in
 
+// tailscaledInit facilitates the lazy initialization of tailscaled.
+// It defers potentially expensive and platform-specific
+// initialization steps until the main function is called and
+// determines that the current process is an actual tailscaled
+// process, rather than a CLI symlink, `tailscaled --version`, or
+// another subcommand that doesn't require full initialization.
+var tailscaledInit lazy.DeferredInit
```

Comment:
I don't want to see more use of lazy.DeferredInit. I'm not a fan of that type.

Plus you just have init-time work and references to it and the code to be deferred necessarily because of its API:

```
func init() {
	tailscaledInit.Defer(func() error {
```

While I don't care about the deadcode eliminiation here in package main, I don't like the pattern for library code.

But if you need to init COM, init COM when you need to. I'd rather seeing that COM was initializing explicitly in code rather than guess which init code in which package might've registered something.